packages/worker/src/isolate-builder.ts

```diff
@@ -66,32 +64,26 @@ export class WorkflowIsolateBuilder {
    *
    * Exports all detected Workflow implementations and some workflow libraries to be used by the Worker.
    */
-  protected async genEntrypoint(vol: typeof memfs.vol, target: string, workflows: string[]): Promise<void> {
-    let code = dedent`
-      const init = require('@temporalio/workflow/lib/init');
-      const internals = require('@temporalio/workflow/lib/internals');
-      internals.state.activityDefaults = ${JSON.stringify(this.activityDefaults)};
-      init.overrideGlobals();
-      const workflows = {};
-      const interceptors = { inbound: [], outbound: [] };
+  protected genEntrypoint(vol: typeof memfs.vol, target: string, workflows: string[]): void {
+    const bundlePaths = [...new Set(workflows.concat(this.workflowInterceptorModules))].map((wf) =>
+      JSON.stringify(path.join(this.workflowsPath, `${wf}.js`))
+    );
+    const code = dedent`
+      const api = require('@temporalio/workflow/lib/worker-interface');
 
-      function addInterceptors({ inbound, outbound }) {
-        interceptors.inbound.push(...inbound);
-        interceptors.outbound.push(...outbound);
-      }
-      export { init, internals, workflows, interceptors };
+      // Bundle all Workflows and interceptor modules for lazy evaluation
+      globalThis.document = api.mockBrowserDocumentForWebpack();
+      // See https://webpack.js.org/api/module-methods/#requireensure
+      require.ensure([${bundlePaths.join(', ')}], function () {});
+      delete globalThis.document;
+
+      api.overrideGlobals();
+      api.setRequireFunc(
+        (name) => require(${JSON.stringify(this.workflowsPath)} + '${path.sep}' + name + '.js')
```

Comment:
Nice :+1: Does this mean for requiring activities, you can do `require('../activities/myActivity')`?

Reply:
No, that's for Workflow internals requiring user code.
It has nothing to do with user code imports.
Let's get rid of the activities alias soon though.